File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    
    let freq = new Array(grid.length * grid[0].length + 1);
    freq.fill(0);
    let a = -1;
    let b = -1;

    grid.forEach(gr => {
        gr.forEach(el =>
        {
            freq[el]++;
        });
    });

    for(let i = 1 ; i < freq.length ; i++)
    {
        if(freq[i] == 2)
            a = i;
        if(freq[i] == 0)
            b = i;
    }
    
    return [a,b];
};

File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        
        int n = grid.length;
        int []freq = new int[n * n + 1];
        int a = -1;
        int b = -1;

        for(int nums[] : grid)
        {
            for(int num : nums)
            {
                freq[num]++;
            }
        }

        for(int i = 1 ; i < freq.length ; i++)
        {
            if(freq[i] == 2)
                a = i;
            if(freq[i] == 0)
                b = i;

            if(a != -1 && b != -1)
                break;
        }

        return new int[]{a,b};
    }
}